src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1606,7 +1607,7 @@ void uct_dc_mlx5_iface_set_ep_failed(uct_dc_mlx5_iface_t *iface,
     }
 
     ep->flags |= UCT_DC_MLX5_EP_FLAG_ERR_HANDLER_INVOKED;
-    uct_rc_fc_reset(&iface->super.super, &ep->fc);
+    uct_rc_iface_fc_reset(&iface->super.super, &ep->fc);
```

Comment:
Actually it looks weird moving to rc_iface, since in DC the FC struct is also per EP.
it should be something like: uct_rc_ep_fc_reset(&ep->super, &ep->fc);

Reply:
it couldn't be done since DC doesn't inherit `uct_rc_ep_t`

Reply:
why not pass `iface` as second argument to `uct_rc_fc_reset`/`uct_rc_fc_init`/`uct_rc_fc_cleanup`?

Reply:
maybe pass window size?

Reply:
but it looks like that fc_init and fc_cleanup could be updated to not allocate/free STATS if FC isn't enabled, wdyt?